src/sentry/tasks/relay.py

```diff
@@ -154,3 +108,162 @@ def schedule_update_config_cache(
+    we want to use primitive types for the arguments.  This is the common validation to make
+    sure only one is provided.
+    """
+    if [bool(organization_id), bool(project_id), bool(public_key)].count(True) != 1:
+        raise TypeError("Must provide exactly one of organzation_id, project_id or public_key")
+
+    if project_id:
+        set_current_event_project(project_id)
+    if organization_id:
+        # Cannot use bind_organization_context here because we do not have a
+        # model and don't want to fetch one
+        sentry_sdk.set_tag("organization_id", organization_id)
+    if public_key:
+        sentry_sdk.set_tag("public_key", public_key)
+
+
+def project_keys_to_update(organization_id=None, project_id=None, public_key=None):
+    """Returns the project keys which need to have their config updated.
+
+    Queries the database for the required project keys.
+    """
+    from sentry.models import Project, ProjectKey
+
+    if organization_id:
+        projects = list(Project.objects.filter(organization_id=organization_id))
+        keys = list(ProjectKey.objects.filter(project__in=projects))
+    elif project_id:
+        projects = [Project.objects.get(id=project_id)]
+        keys = list(ProjectKey.objects.filter(project__in=projects))
+    elif public_key:
+        try:
+            keys = [ProjectKey.objects.get(public_key=public_key)]
+        except ProjectKey.DoesNotExist:
+            # In this particular case, where a project key got deleted and
+            # triggered an update, we know that key doesn't exist and we want to
+            # avoid creating more tasks for it.
+            #
+            # In other similar cases, like an org being deleted, we potentially
+            # cannot find any keys anymore, so we don't know which cache keys
+            # to delete.
+            projectconfig_cache.set_many({public_key: {"disabled": True}})
+            keys = []
+
+    else:
+        assert False
+
+    return keys
+
+
+def compute_project_configs(project_keys):
+    """Computes the project configs for all given project keys."""
+    from sentry.models import ProjectKeyStatus
+    from sentry.relay.config import get_project_config
+
+    config_cache = {}
+    for key in project_keys:
+        if key.status != ProjectKeyStatus.ACTIVE:
+            project_config = {"disabled": True}
+        else:
+            project_config = get_project_config(
+                key.project, project_keys=[key], full_config=True
+            ).to_dict()
+        config_cache[key.public_key] = project_config
+
+    projectconfig_cache.set_many(config_cache)
+
+
+@instrumented_task(
+    name="sentry.tasks.relay.invalidate_project_config",
+    queue="relay_config",
+    acks_late=True,
+    soft_time_limit=30,
+    time_limit=32,
+)
+def invalidate_project_config(
+    organization_id=None, project_id=None, public_key=None, update_reason="invalidated", **kwargs
```

Comment:
Since we're now using `trigger`:
```suggestion
    organization_id=None, project_id=None, public_key=None, trigger="invalidated", **kwargs
```